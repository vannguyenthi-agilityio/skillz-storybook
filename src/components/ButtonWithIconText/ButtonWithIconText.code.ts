export const btnWithIconCreate = `
import ButtonWithIconText from '.';

/**
 * The used to render ButtonWithIconText with action type Create.
 *
 */

 <ButtonWithIconText />
`;

export const btnWithIconUpload = `
import ButtonWithIconText from '.';
import { UploadIcon } from '@components';

/**
 * The used to render ButtonWithIconText with action type Upload.
 *
 */

 <ButtonWithIconText
  textAction='Upload'
  textType='an exam'
  icon={<UploadIcon />}
/>
`;

export const btnWithIconUploadIsAcive = `
import ButtonWithIconText from '.';
import { UploadIcon } from '@components';

/**
 * The used to render ButtonWithIconText with action type Upload isActive.
 *
 */

 <ButtonWithIconText
  textAction='Upload'
  textType='an exam'
  valueTokens='20',
  isActive=true
  variant='ghost'
  icon={<UploadIcon />}
/>
`;

export const btnWithIconUploadWithValueTokens = `
import ButtonWithIconText from '.';
import { UploadIcon } from '@components';

/**
 * The used to render ButtonWithIconText with action type Upload with ValueTokens.
 *
 */

 <ButtonWithIconText
  textAction='Upload'
  textType='an exam'
  valueTokens='20',
  icon={<UploadIcon />}
  />
`;

export const btnWithIconInput = `
import ButtonWithIconText from '.';
import { InputIcon } from '@components';

/**
 * The used to render ButtonWithIconText with action type Input.
 *
 */

 <ButtonWithIconText
  textAction='Input'
  textType='an exam'
  icon={<InputIcon />}
/>
`;
