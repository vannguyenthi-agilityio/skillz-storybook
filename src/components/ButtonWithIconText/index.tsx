// Components
import { Flex, Button, Text } from '@chakra-ui/react';
import { CreateIcon, StarLayoutIcon } from '@components';

// Enums
import { COLORS } from '@enums';

// Helpers
import { convertPxToRem } from '@helpers/common';

// Themes
import CUSTOM_COLORS_PALETTE from '@themes/colors';

// Interface
import { IButtonWithIconText } from '@interfaces/buttonWithIconText.interface';

export const ButtonWithIconText = ({
  valueTokens = '',
  colorScheme = `${COLORS.CYAN}.500`,
  bgColorBoxText = `${COLORS.CYAN}.400`,
  icon = <CreateIcon />,
  textAction = 'Create',
  textType = 'an examination',
  isActive,
  ...rest
}: IButtonWithIconText) => {
  return (
    <Button
      alignItems='center'
      data-testid={`btn-${textAction}`}
      bg={colorScheme}
      w={convertPxToRem(190)}
      h={convertPxToRem(280)}
      justifyContent='center'
      p='0'
      isActive={isActive}
      borderRadius='xl'
      flexDirection='column'
      onClick={rest.onClick}
      {...rest}
    >
      {valueTokens && (
        <Flex
          w='full'
          alignItems='center'
          justifyContent='end'
          mb='4'
        >
          <StarLayoutIcon
            w={convertPxToRem(24)}
            h={convertPxToRem(24)}
            bgColor={isActive ? CUSTOM_COLORS_PALETTE.orange[500] : CUSTOM_COLORS_PALETTE.yellow[750]}
          />
          <Text
            px='2'
            fontSize='lg'
            color={isActive ? 'black' : 'yellow.750'}
          >
            {valueTokens}
          </Text>
        </Flex>
      )}
      <Flex
        alignItems='center'
        justifyContent='center'
        aria-label={`Icon ${textAction}`}
        borderColor='grey.850'
        borderStyle='solid'
        borderWidth='2px'
        borderRadius='full'
        w={convertPxToRem(80)}
        h={convertPxToRem(80)}
      >
        {icon}
      </Flex>
      <Flex
        flexDirection='column'
        textAlign='center'
        w='full'
        mb='4'
        py={convertPxToRem(7)}
        mt={convertPxToRem(31)}
        color='grey.850'
        backgroundColor={`${bgColorBoxText}`}
      >
        <Text
          fontSize='4xl'
          fontWeight='300'
          lineHeight={convertPxToRem(40)}
        >
          {textAction}
        </Text>
        <Text
          fontSize='xl'
          fontWeight='300'
          lineHeight={convertPxToRem(36)}
        >
          {textType}
        </Text>
      </Flex>
    </Button>
  );
};

export default ButtonWithIconText;
