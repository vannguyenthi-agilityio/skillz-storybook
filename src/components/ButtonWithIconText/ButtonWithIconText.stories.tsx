import Storybook from '@storybook/react';

// Components
import ButtonWithIconText from '.';
import { UploadIcon, InputIcon } from '@components';

// Code
import {
  btnWithIconInput,
  btnWithIconUpload,
  btnWithIconCreate,
  btnWithIconUploadIsAcive,
  btnWithIconUploadWithValueTokens
} from './ButtonWithIconText.code';

export default {
  title: 'Components/ButtonWithIconText',
  component: ButtonWithIconText,
  parameters: {
    componentSource: {
      language: 'typescript'
    }
  },
  argTypes: {
    colorScheme: {
      options: ['yellow.300', 'cyan.400', 'orange.400'],
      control: {
        type: 'select'
      },
      defaultValue: 'cyan.400'
    },
    bgColorBoxText: {
      options: ['yellow.500', 'cyan.500', 'orange.500'],
      control: {
        type: 'select'
      },
      defaultValue: 'cyan.500'
    }
  }
} as Storybook.ComponentMeta<typeof ButtonWithIconText>;

const Template: Storybook.ComponentStory<typeof ButtonWithIconText> = (args) => {
  return <ButtonWithIconText {...args} />;
};

export const ButtonWithIconTextCreate = Template.bind({});
ButtonWithIconTextCreate.args = {
  textAction: 'Create'
};
ButtonWithIconTextCreate.parameters = {
  componentSource: {
    code: btnWithIconCreate
  },
  argTypes: {
    colorScheme: {
      defaultValue: 'yellow'
    }
  }
};

export const ButtonWithIconTextInput = Template.bind({});
ButtonWithIconTextInput.args = {
  textAction: 'Input',
  textType: 'an exam',
  icon: <InputIcon />
};
ButtonWithIconTextInput.parameters = {
  componentSource: {
    code: btnWithIconInput
  },
  argTypes: {
    colorScheme: {
      defaultValue: 'blue'
    }
  }
};

export const ButtonWithIconTextUpload = Template.bind({});
ButtonWithIconTextUpload.args = {
  textAction: 'Upload',
  textType: 'an exam',
  icon: <UploadIcon />
};
ButtonWithIconTextUpload.parameters = {
  componentSource: {
    code: btnWithIconUpload
  },
  argTypes: {
    colorScheme: {
      defaultValue: 'orange'
    }
  }
};

export const ButtonWithIconTextUploadWithValueTokens = Template.bind({});
ButtonWithIconTextUploadWithValueTokens.args = {
  textAction: 'Upload',
  textType: 'an exam',
  valueTokens: '20',
  icon: <UploadIcon />
};
ButtonWithIconTextUploadWithValueTokens.parameters = {
  componentSource: {
    code: btnWithIconUploadWithValueTokens
  }
};

export const ButtonWithIconTextUploadIsActive = Template.bind({});
ButtonWithIconTextUploadIsActive.args = {
  textAction: 'Upload',
  textType: 'an exam',
  icon: <UploadIcon />,
  isActive: true,
  valueTokens: '20',
  variant: 'ghost'
};
ButtonWithIconTextUploadIsActive.parameters = {
  componentSource: {
    code: btnWithIconUploadIsAcive
  },
  argTypes: {
    colorScheme: {
      defaultValue: 'orange'
    }
  }
};
