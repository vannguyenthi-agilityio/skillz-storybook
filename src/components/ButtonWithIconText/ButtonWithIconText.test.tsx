// Libraries
import { render } from '@testing-library/react';

// Components
import ButtonWithIconText from '.';

describe('ButtonWithIconText Component type create', () => {
  test('should render correctly', () => {
    const { container } = render(<ButtonWithIconText />);

    const btnWithIconTextCreate = container.querySelector('.chakra-button');
    expect(btnWithIconTextCreate).toMatchSnapshot();
  });
});

it('should render ButtonWithIconText with type input', () => {
  const { getByText } = render(<ButtonWithIconText textAction='Input' />);

  const button = getByText('Input');
  expect(button).toBeTruthy();
});

it('should render ButtonWithIconText with type upload', () => {
  const { getByText } = render(<ButtonWithIconText textAction='Upload' />);

  const button = getByText('Upload');
  expect(button).toBeTruthy();
});
