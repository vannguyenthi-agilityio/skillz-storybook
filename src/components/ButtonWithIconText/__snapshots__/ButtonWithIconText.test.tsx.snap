// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWithIconText Component type create should render correctly 1`] = `
<button
  class="chakra-button css-6pn8h7"
  data-testid="btn-Create"
  type="button"
>
  <div
    aria-label="Icon Create"
    class="css-lzc16y"
  >
    <svg
      fill="none"
      height="17"
      viewBox="0 0 17 17"
      width="17"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.68748 0.375011H9.40998V7.68751H16.7225V9.41001H9.40998V16.7225H7.68748V9.41001H0.374985V7.68751H7.68748V0.375011Z"
        fill="black"
      />
    </svg>
  </div>
  <div
    class="css-6t42k7"
  >
    <p
      class="chakra-text css-168o111"
    >
      Create
    </p>
    <p
      class="chakra-text css-8tawpc"
    >
      an examination
    </p>
  </div>
</button>
`;
